src/main/java/com/ksabbak/javaserver/server/request/RequestReader.java

```diff
@@ -0,0 +1,71 @@
+package com.ksabbak.javaserver.server.request;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RequestReader {
```

Comment:
Having a `RequestReader` makes a lot of sense.

Given that it's multiple steps to parse the request, introducing a `RequestBuilder` might help clean the code up. You could parse the info from the first "part" of the request, call the appropriate builder functions, then use the content length to parse the body, then finally call `build`

Reply:
I'll try to make that a thing. I noticed it moving in that direction but hadn't been sure enough to fully commit.